add note about unit stride to layout

artv3 opened the pull request:
# Summary
Add a note in the documentation regarding unit stride for RAJA::layouts

kab163 commented on docs/sphinx/user_guide/tutorial/view_layout.rst:
```diff
@@ -210,36 +221,36 @@ for linear index 7, you can do::
-methods are in bounds to avoid accessing data outside 
-of the View or computing invalid indices. 
+index computation methods. Therefore, it is the responsibility of a user
+to ensure that indices passed to ``RAJA::View`` and ``RAJA::Layoout``
```
Typo on Layout

artv3 replied:
```suggestion
to ensure that indices passed to ``RAJA::View`` and ``RAJA::Layout``
```

kab163 commented on docs/sphinx/user_guide/tutorial/view_layout.rst:
```diff
@@ -55,20 +55,31 @@ which we define as:
    :language: C++
 
```
Maybe we can add a warning about RAJA View performance?
```suggestion
.. warning:: In order to get the best performance out of ``RAJA::View``, be sure you are building in release mode (i.e. make sure you don't have a debug build when doing performance comparisons).
```
¯\_(ツ)_/¯

artv3 replied:
If you use -G in a Cuda build it removes kernel optimizations

adayton1 replied:
But -lineinfo should not, right?

artv3 replied:
Right

adayton1 replied:
OK, I leave it up to you whether you want to add this note.

artv3 replied:
Yes, that is a good point. We can add it somewhere else.

adayton1 reviewed (approved):
LGTM

rhornung67 commented on docs/sphinx/user_guide/tutorial/view_layout.rst:
```diff
@@ -55,20 +55,31 @@ which we define as:
+
+            RAJA::Layout<2, int, 1>
+
+          In the example above index 1 will be marked to have unit stride making
```
The note is correct in that it correctly identifies the stride-1 index in the 2-dim'l layout case. The default stride-1 index in our convention is the rightmost index. I would suggestion adding that the third argument is optional.  Also, I would use a permuted layout example to illustrate the optimized use case to avoid extraneous indexing arithmetic.

artv3 replied:
Resolved as we discussed at the meeting that the default is  -1 marking no index as containing unit stride.

artv3 commented:
TODO: Add 3D permuted layout example

artv3 commented:
@rhornung67 , I went through and added the unit indexing on our RAJA View exercise and referenced it in the docs as an example of doing 3D permutations and unit indexing. I think I've addressed all feedback now.
